sycl/test/basic_tests/macros.cpp

```diff
@@ -1,18 +1,14 @@
-// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
-//==------------------- macros.cpp - SYCL buffer basic test ----------------==//
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple -fsycl-device-only -dM -E %s -o %t.device
+// RUN: %clangxx %fsycl-host-only -dM -E %s -o %t.host
 //
-// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
-// See https://llvm.org/LICENSE.txt for license information.
-// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+// RUN: FileCheck %s < %t.device --check-prefixes=DEVICE,COMMON \
+// RUN:     --implicit-check-not=__SPIRV
+// RUN: FileCheck %s < %t.host --check-prefixes=COMMON \
+// RUN:      --implicit-check-not=__SPIRV
```

Comment:
That would be a huge effort, because at the moment we export at least 60 `__SYCL*` macro from our headers and ~40 `SYCL*` macro.

I'm sure that some of them should be exported in accordance with the [spec](https://www.khronos.org/registry/SYCL/specs/sycl-2020/html/sycl-2020.html#_preprocessor_directives_and_macros).

Will it be acceptable to perform `__SYCL*` macro cleanup as a separate PR to make this one smaller? The thing with `__SYCL*` macro is that they sometimes defined in one header, but then used in another (or in a `.cpp` file), which makes them harder to undefine (need to find the right place for that)

Reply:
Limited scope of the PR to only `__SPIRV*` macro in 3d1b313fabbd8dabf70460b65102ed3fb22afcce